homeassistant/components/avreceiver/media_player.py

```diff
@@ -0,0 +1,201 @@
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_OFF, STATE_ON
+
+from .const import (
+    CONF_ZONE1,
+    DOMAIN as AVRECEIVER_DOMAIN,
+    SIGNAL_AVR_UPDATED,
+    THREE_DECIBELS,
+)
+
+BASE_SUPPORTED_FEATURES = (
+    SUPPORT_VOLUME_MUTE
+    | SUPPORT_VOLUME_SET
+    | SUPPORT_VOLUME_STEP
+    | SUPPORT_SELECT_SOURCE
+    | SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+)
+
+
+async def async_setup_entry(hass, entry: ConfigEntry, async_add_entities):
+    """Add media players for a config entry."""
+    zones = hass.data[AVRECEIVER_DOMAIN][entry.entry_id][DOMAIN]
+    entities = []
+    for name, zone in zones.items():
+        await zone.update_all()
+        if name == CONF_ZONE1:
+            entities.append(AVRMainZone(zone, name))
+        else:
+            entities.append(AVRZone(zone, name))
+    async_add_entities(entities)
+
+
+class AVRZone(MediaPlayerEntity):
+    """Representation of an AV Receiver zone."""
+
+    def __init__(self, zone: Zone, zone_name: str):
+        """Initialize the device."""
+        self._zone = zone
+        self._zone_name = zone_name
+        self._avr = zone._avr  # type: AVReceiver
+        self._signals = []
+        self._supported_features_base = BASE_SUPPORTED_FEATURES
+        self._volume_multiplier = (
+            self._zone.max_volume or 0 - self._zone.min_volume or -80
+        )
+
+    async def _avr_updated(self):
+        """Handle avr attribute updated."""
+        await self.async_update_ha_state(True)
+
+    async def async_added_to_hass(self):
+        """Device added to hass."""
+        # Update state when avr changes
+        self._signals.append(
+            self.hass.helpers.dispatcher.async_dispatcher_connect(
+                SIGNAL_AVR_UPDATED, self._avr_updated
+            )
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if the device is available."""
+        return self._avr.connection_state == "connected"
+
+    @property
+    def device_info(self) -> dict:
+        """Get attributes about the device."""
+        return {
+            "identifiers": {(AVRECEIVER_DOMAIN, self._avr.host)},
+            "name": self._avr.friendly_name,
+            "model": self._avr.model,
+            "manufacturer": self._avr.manufacturer,
+        }
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"{self._avr.friendly_name} {self._zone_name}"
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return STATE_ON if self._zone.power else STATE_OFF
+
+    @property
+    def is_volume_muted(self):
+        """Return boolean if volume is currently muted."""
+        return self._zone.mute
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return (
+            f"{AVRECEIVER_DOMAIN}-"
+            f"{self._avr.serial_number or self._avr.mac or self._avr.host}-"
```

Comment:
unique_id needs to always be the same for a device, therefore do not use host in unique_id because it can change.
Besides I would not use the `or`
serial_number or mac is enough, just pick one.